app/services/criteria_matcher.py

```
@@ -309,6 +296,32 @@ def _match_group(gid: int, value: str) -> bool:
+
+        try:
+            # Try parsing with timezone first (e.g., "2024-01-15 10:30:00 +0000")
+            if "+" in date_str or (date_str.count("-") >= 3 and date_str[-5] in "+-"):
+                try:
+                    dt = datetime.strptime(date_str, "%Y-%m-%d %H:%M:%S %z")
+                    return dt.timestamp()
+                except ValueError:
+                    # Fallback for failing %z: try manual UTC parsing
+                    parts = date_str.rsplit(" ", 1)
+                    if len(parts) == 2:
+                        dt_naive = datetime.strptime(parts[0], "%Y-%m-%d %H:%M:%S")
```

Ruff (0.15.9)

[warning] Naive datetime constructed using `datetime.datetime.strptime()` without %z

(DTZ007)